### DISCUSSION TITLE
Doubt about getCellDependents

### DISCUSSION THREAD

**Discussion by nathipg** [handsontable/hyperformula#929] [API Questions]
Hello!

I'm using hyperformula version 1.3.1 and I'm having a strange behavior from getCellDependents.

I made this sample to show it: https://jsfiddle.net/r1q2nbc9/1/

Based on the sample, cell E12 has this formula:

`=SUMIFS(E2:E9,A2:A9,A12,B2:B9,B12,C2:C9,C12)`

If I run the following command to get the dependents of E2:

`
formulaPlugin.engine.getCellDependents({
  sheet: 0,
  col: 4,
  row: 1,
});
`

I have the this return:

![image](https://user-images.githubusercontent.com/14853904/157304978-b5f565c3-3c2e-41fe-b47a-f154b21050dd.png)

Which is a range from E2 to E9, why is this the function return?

To me, the return should be an array with the address to E12 cell.

What am I doing wrong?

**Comment by sequba** [comment]
Hi @nathipg!

It may seem illogical but this is the intended behaviour of `getCellDependents`. Most ranges are kept in the dependency graph as separate nodes for performance reasons. [Read more about ranges in the dependency graph](https://handsontable.github.io/hyperformula/guide/key-concepts.html#handling-ranges)

In your case, you can run `getCellDependents` twice to get the address of E12 and E13 like this:

```
const level1Dependents = formulaPlugin.engine.getCellDependents({
  sheet: 0,
  col: 4,
  row: 1,
});

const level2Dependents = formulaPlugin.engine.getCellDependents(level1Dependents[0]);

console.log('Level 1 dependents:', level1Dependents);
console.log('Level 2 dependents:', level2Dependents);
```
Prints:
```
"Level 1 dependents:", [{
  end: {
    col: 4,
    row: 8,
    sheet: 0
  },
  start: {
    col: 4,
    row: 1,
    sheet: 0
  }
}]
"Level 2 dependents:", [{
  col: 4,
  row: 11,
  sheet: 0
}, {
  col: 4,
  row: 12,
  sheet: 0
}]
```

Does it solve your issue?

**Reply by nathipg** [reply]
Thank you! That solved my problem.
One question, is it possible to have more than 2 level dependents?

**Reply by sequba** [reply]
Yes, it is. If you want to get all the dependants of a cell you can use the BFS-like algorithm as described here: https://github.com/handsontable/hyperformula/discussions/889#discussioncomment-2078320

**Reply by nathipg** [reply]
Thanks! I'm gonna see it

**Comment by sequba** [comment]
Related issues for the future reference:
https://github.com/handsontable/hyperformula/issues/934
https://github.com/handsontable/hyperformula/issues/930
#931

**Comment by sequba** [comment]
@nathipg A thorough explanation of these functions can be found in the [Dependency Graph guide](https://handsontable.github.io/hyperformula/guide/dependency-graph.html) and API reference for [getCellDependents](https://handsontable.github.io/hyperformula/api/classes/hyperformula.html#getcelldependents) and [getCellPrecedents](https://handsontable.github.io/hyperformula/api/classes/hyperformula.html#getcellprecedents).
